Source/CSharpEssentials/Extensions.cs

```diff
@@ -168,9 +170,36 @@ private static bool IsGeneratedCode(string filePath)
             return false;
         }
 
-        public static bool IsGeneratedCode(this SyntaxTree tree)
+        public static bool IsGeneratedCode(this SyntaxNode node)
         {
-            return IsGeneratedCode(tree.FilePath);
+            return IsInGeneratedCodeContext(node)
+                || IsGeneratedCode(node.SyntaxTree.FilePath);
+        }
+
+        private static bool IsInGeneratedCodeContext(this SyntaxNode node)
+        {
+            do
+            {
+                var method = node as MethodDeclarationSyntax;
+                if (method != null && ContainsGeneratedCodeAttribute(method.AttributeLists))
+                    return true;
```

Comment:
What about fields initializers, properties, indexers, operators, and conversions?